language/move-prover/stackless-bytecode-generator/src/annotations.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    any::{Any, TypeId},
+    collections::BTreeMap,
+};
+
+/// A container for an extensible, dynamically typed set of annotations.
+#[derive(Debug, Default)]
+pub struct Annotations {
+    map: BTreeMap<TypeId, Box<dyn Any>>,
```

Comment:
I think Any is the only way to do this, because it has actually dynamic type information and allows for save runtime tests and casts. A trait cannot be downcasted unless you implement your own dynamic typing. This is an often discussed restriction of Rusts type system. You can also not downcast a Dyn type to trait, but only to a concrete type.

If a trait would work here, that would be nice because there is at least one common thing which would be useful: debug printing for baseline tests (followup PR). But that's about it which would be common, as the annotation can really be arbitrary, and is interpreted only by code which nows the fully concrete type T.